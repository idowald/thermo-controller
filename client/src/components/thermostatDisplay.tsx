import * as React from "react";
import { simpleTimeFormat } from "../utils/utils";
import AcUnitIcon from "@material-ui/icons/AcUnit";
import WhatshotIcon from "@material-ui/icons/Whatshot";
import FlagIcon from "@material-ui/icons/Flag";
import ScheduleIcon from "@material-ui/icons/Schedule";
import { createStyles, makeStyles } from "@material-ui/core/styles";
import type {thermostatClient} from "../store/reducers/reducer";
const useStyles = makeStyles(() =>
  createStyles({
    hidden: {
      display: "none"
    },
    thermoDisplay: {
      width: "300px",
      display: "flex",
      alignItems: "center",
      justifyContent: "space-evenly"
    },
      spans: {
          display: "flex",
          alignItems: "flex-end",
      }
  })
);
export const ThermostatDisplay = ({
  currentTemp,
  currentSetpoint,
  timestamp
}: thermostatClient) => {
  const classes = useStyles();
  const timeFormat = timestamp? simpleTimeFormat(new Date(timestamp)) : "00:00.00";
  return (
    <div className={classes.thermoDisplay}>
      <span className={classes.spans}>
        <AcUnitIcon className={currentTemp <= 23 ? "" : classes.hidden} />{" "}
        <WhatshotIcon className={currentTemp > 23 ? "" : classes.hidden} />{" "}
        {currentTemp}°
      </span>
      <span className={classes.spans}>
        <FlagIcon /> {currentSetpoint}°{" "}
      </span>
      <span className={classes.spans}>
        <ScheduleIcon />
        {timeFormat}
      </span>
    </div>
  );
};
