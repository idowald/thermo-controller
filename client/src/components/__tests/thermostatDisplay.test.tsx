import React from "react";
import renderer from "react-test-renderer";
import {ThermostatDisplay} from "../thermostatDisplay";

test("test thermo display for correct numbers", () => {
  const timestamp = new Date().getTime();
  const currentSetpoint = 26.31;
  const currentTemp =22.31;
  const rendered = renderer.create(
    <ThermostatDisplay timestamp={timestamp} currentSetpoint={currentSetpoint} currentTemp={22.31} />
  );
  const renderJson = rendered.toJSON();
  // @ts-ignore
  expect(renderJson.children[0].children[4]).toEqual(currentTemp.toString());
  // @ts-ignore
  expect(renderJson.children[1].children[2]).toEqual(currentSetpoint.toString());

});
test("test thermo display snapshot", () => {
  const timestamp = new Date(1232224).getTime();
  const currentSetpoint = 26.31;
  const currentTemp =22.31;
  const rendered = renderer.create(
      <ThermostatDisplay timestamp={timestamp} currentSetpoint={currentSetpoint} currentTemp={22.31} />
  );
  expect(rendered).toMatchSnapshot();
});
