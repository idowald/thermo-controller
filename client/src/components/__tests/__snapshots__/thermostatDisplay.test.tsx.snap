// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test thermo display snapshot 1`] = `
<div
  className="makeStyles-thermoDisplay-2"
>
  <span
    className="makeStyles-spans-3"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
      />
    </svg>
     
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-hidden-1"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
      />
    </svg>
     
    22.31
    °
  </span>
  <span
    className="makeStyles-spans-3"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14.4 6L14 4H5v17h2v-7h5.6l.4 2h7V6z"
      />
    </svg>
     
    26.31
    °
     
  </span>
  <span
    className="makeStyles-spans-3"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    01:20.32
  </span>
</div>
`;
